### PR TITLE
Download assessment items on setup

### PR DESCRIPTION
**Pull request by aronasorman**
Solves #https://github.com/learningequality/ka-lite/issues/3045. 

Summary of changes:
1. There is now a new file, `assessmentitems.json.version` included on every assessment items zip file.
2. This file is extracted to `data/khan/`, along with `assessmentitems.json`.
3. When `unpack_assessment_zip` is called, this file is checked. If it's not present (implying no assessment item resources are downloaded), or it's less than the KA Lite version number, the latest assessment items zip is downloaded (we may want to version the zip file name here).
4. `unpack_assessment_zip` is now called in the `setup` management command.

### CODE REVIEW

File: kalite/contentload/management/commands/generate_assessment_zips.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def handle_noargs(self, **options):
         # without redownloading all files. Not possible currently because ZipFile has no `delete`.
         logging.info("downloading images")
         with open(ZIP_FILE_PATH, "w") as f:
-            zf = zipfile.ZipFile(ZIP_FILE_PATH, "w")  # zipfile.ZipFile isn't a context manager yet for python 2.6
+            zf = zipfile.ZipFile(f, "w")  # zipfile.ZipFile isn't a context manager yet for python 2.6
```
Comment (aronasorman) [inline kalite/contentload/management/commands/generate_assessment_zips.py]: Just cleaning up a code smell I did here, which is not use the `f` I just opened.

File: kalite/contentload/management/commands/unpack_assessment_zip.py [hunk 2]
```diff
@@ -20,17 +26,30 @@
...
 
 ASSESSMENT_ITEMS_PATH = os.path.join(KHAN_DATA_PATH, "assessmentitems.json")
+ASSESSMENT_ITEMS_VERSION_PATH = os.path.join(KHAN_DATA_PATH, "assessmentitems.json.version")
 
 logging = settings.LOG
 
 class Command(BaseCommand):
 
+    option_list = BaseCommand.option_list + (
+        make_option("-f", "--force",
```
Comment (aronasorman) [inline kalite/contentload/management/commands/unpack_assessment_zip.py]: If this flag isn't specified, we will always check if we need to download the latest assessment zip file.

File: .gitignore [hunk 3]
```diff
@@ -58,3 +58,4 @@ ghostdriver.log
 /kalite/_built.touch
 /kalite/contentload/tests/fixtures/
 *.swp
+/data/khan/assessmentitems.json.version
```
Comment (aronasorman) [inline .gitignore]: `assessmentitems.json.version` has the KA Lite version of the software that generated this zip file.
a.k.a
"0.13"

File: kalite/contentload/tests/unpack_assessment_zip.py [hunk 4]
```diff
@@ -10,21 +11,47 @@
...
+    def setUp(self):
+        _, self.zipfile_path = tempfile.mkstemp()
+        with open(self.zipfile_path, "w") as f:
+            zf = zipfile.ZipFile(f, "w")
+            for dirpath, _, filenames in os.walk(os.path.join(os.path.dirname(__file__), "fixtures")):
+                # this toplevel for loop should only do one loop, but
+                # it's in a for loop nonetheless since it's more idiomatic
+                for filename in filenames:
+                    full_path = os.path.join(dirpath, filename)
+                    zf.write(full_path, filename)
```
Comment (aronasorman) [inline kalite/contentload/tests/unpack_assessment_zip.py]: For these tests, I used to have a zipfile committed in git inside `fixtures/`. Now I just generate this zip on the fly using the contents found in `fixtures/`.

File: kalite/contentload/tests/unpack_assessment_zip.py [hunk 5]
```diff
@@ -10,21 +11,47 @@
...
+        upgrade_method.return_value = False
+
+        # test that we don't update when given a url
+        url = "http://fakeurl.com/test.zip"
+        call_command("unpack_assessment_zip", url)
+        self.assertEqual(get_method.call_count, 0, "requests.get was called even if we should've skipped!")
+
+        filename = "/fake/file/somewhere.zip"
+        call_command("unpack_assessment_zip", filename)
+        self.assertEqual(mod.open.call_count, 0,  "open was called even if we should've skipped!")
```
Comment (aronasorman) [inline kalite/contentload/tests/unpack_assessment_zip.py]: I mock the open function, and make sure it's not being called.

Comment (aronasorman) [inline kalite/contentload/tests/unpack_assessment_zip.py]: a.k.a. we never get to that portion of the code.

File: kalite/contentload/tests/unpack_assessment_zip.py [hunk 6]
```diff
@@ -10,21 +11,47 @@
...
+
+
     @patch.object(requests, "get", autospec=True)
     def test_command_with_url(self, get_method):
         url = "http://fakeurl.com/test.zip"
 
-        with open(ASSESSMENT_ZIP_SAMPLE_PATH) as f:
+        with open(self.zipfile_path) as f:
```
Comment (aronasorman) [inline kalite/contentload/tests/unpack_assessment_zip.py]: We no longer use the hardcoded zip in `fixtures/`.

File: kalite/contentload/tests/unpack_assessment_zip.py [hunk 7]
```diff
@@ -55,6 +82,21 @@ def test_command_with_local_path(self):
...
+            for dirpath, _, filenames in os.walk(os.path.join(os.path.dirname(__file__), "fixtures")):
+                # this toplevel for loop should only do one loop, but
+                # it's in a for loop nonetheless since it's more idiomatic
+                for filename in filenames:
+                    full_path = os.path.join(dirpath, filename)
+                    zf.write(full_path, filename)
+            zf.close()
+
+    def tearDown(self):
+        os.unlink(self.zipfile_path)
```
Comment (aronasorman) [inline kalite/contentload/tests/unpack_assessment_zip.py]: Don't forget to delete the zip we just created.

File: kalite/distributed/management/commands/setup.py [hunk 8]
```diff
@@ -251,6 +250,10 @@ def handle(self, *args, **options):
         call_command("syncdb", interactive=False, verbosity=options.get("verbosity"))
         call_command("migrate", merge=True, verbosity=options.get("verbosity"))
 
+        # download assessment items
+        # TODO-BLOCKER (aron): do not hardcode this, and put in the proper URL once we have a redirect set up
+        call_command("unpack_assessment_zip", "http://eslgenie.com/media/assessment_item_resources.zip")
```
Comment (aronasorman) [inline kalite/distributed/management/commands/setup.py]: We'll want to change this to the proper url sometime before release.

**Comment by aronasorman** [comment]
Merging this in would break the assessment zip unpacking, since the zip file we serve doesn't have `assessmentitems.json.version`. We can just manually insert that file though with `"0.13"` as its contents.

**Comment by aronasorman** [comment]
I've strived to make the commits self-contained, so I think that's the best way to review this PR.

**Comment by aronasorman** [comment]
Weird travis is weird. My latest commit is https://github.com/aronasorman/ka-lite/commit/be9fa6302711d3c390a45bc2004d049fd90063b3, yet it's testing 62ce8d9. How am I gonna make this test pass?!

**Comment by rtibbles** [comment]
Weird, it is. Nothing jumps out at me from the code as a cause for concern. Will try to get the tests to work.

**Comment by rtibbles** [comment]
The commit https://github.com/learningequality/ka-lite/commit/62ce8d9cbf9e5371c91917fe15470734a3265f58 is the merge commit of this into develop, I think.

**Comment by rtibbles** [comment]
Again, this is now failing due to timeouts on Python 2.6.

**Comment by rtibbles** [comment]
@aronasorman I fixed the tests that were breaking previously. The only issue was a missing assessmentitems.json.version from the fixtures folder.

**Pull request merged**